DSCResources/MSFT_xNetBIOS/MSFT_xNetBIOS.psm1

```diff
@@ -0,0 +1,112 @@
+Add-Type -TypeDefinition @'
+   public enum NetBiosSetting
+   {
+      Default,
+      Enable,
+      Disable
+   }
+'@
+
+
+function Get-TargetResource
+{
+    [CmdletBinding()]
+    [OutputType([System.Collections.Hashtable])]
+    param
+    (
+        [parameter(Mandatory = $true)]
+        [System.String]
+        $InterfaceAlias,
+
+        [parameter(Mandatory = $true)]
+        [ValidateSet("Default","Enable","Disable")]
+        [System.String]
+        $Setting
+    )
+    try
+    {
+        $NetAdapterConfig = Get-CimInstance -ClassName Win32_NetworkAdapter -Filter "NetConnectionID=`"$InterfaceAlias`"" | Get-CimAssociatedInstance -ResultClassName Win32_NetworkAdapterConfiguration
+    }
+    catch
+    {
+        Write-Error -ErrorRecord $_ -ErrorAction Stop
+    }
+    return @{
+        InterfaceAlias = $InterfaceAlias
+        Setting = $([NETBIOSSetting].GetEnumValues()[$NetAdapterConfig.TcpipNetbiosOptions])
+    }
+}
+
+
+function Set-TargetResource
+{
+    [CmdletBinding()]
+    param
+    (
+        [parameter(Mandatory = $true)]
+        [System.String]
+        $InterfaceAlias,
+
+        [parameter(Mandatory = $true)]
+        [ValidateSet("Default","Enable","Disable")]
+        [System.String]
+        $Setting
+    )
+    $NetAdapterConfig = Get-CimInstance -ClassName Win32_NetworkAdapter -Filter "NetConnectionID=`"$InterfaceAlias`"" | Get-CimAssociatedInstance -ResultClassName Win32_NetworkAdapterConfiguration
+    if ($Setting -eq [NETBIOSSetting]::Default) 
+    {
+        #If DHCP is not enabled, settcpipnetbios CIM Method won't take 0 so overwrite registry entry instead.
+        $null = Set-ItemProperty -Path "HKLM:\SYSTEM\CurrentControlSet\Services\NetBT\Parameters\Interfaces\Tcpip_$($NetAdapterConfig.SettingID)" -Name NetbiosOptions -Value 0
+    }
+    else
+    {
+        $null = $NetAdapterConfig | Invoke-CimMethod -MethodName settcpipnetbios -Arguments @{TcpipNetbiosOptions = [uint32][NETBIOSSetting]::$Setting.value__}
+    }
+}
+
+
+function Test-TargetResource
+{
+    [CmdletBinding()]
+    [OutputType([System.Boolean])]
+    param
+    (
+        [parameter(Mandatory = $true)]
+        [System.String]
+        $InterfaceAlias,
+
+        [parameter(Mandatory = $true)]
+        [ValidateSet("Default","Enable","Disable")]
+        [System.String]
+        $Setting
+    )
+    $NIC = Get-CimInstance -ClassName Win32_NetworkAdapter -Filter "NetConnectionID=`"$InterfaceAlias`""
+    if ($Null -ne $NIC)
+    {
+        Write-Verbose -Message "Interface $InterfaceAlias detected with Index number: $($NIC.InterfaceIndex)"
+    }
+    else
+    {
+        Write-Error -Message "Nic with Alias $InterfaceAlias was not found" -ErrorAction Stop

```

Comment:
You'll want to throw an error object here instead of using Write-Error, this makes it easier to track down exactly where we've hit a problem and allows us to write tests to easily capture the error objects.

See [xFirewall](https://github.com/PowerShell/xNetworking/blob/dev/DSCResources/MSFT_xFirewall/MSFT_xFirewall.psm1#L707) for an example of this.